dom.go

```diff
@@ -1527,6 +1528,7 @@ type Element interface {
 	GetElementsByTagNameNS(ns string, name string) []Element
 	HasAttribute(string) bool
 	HasAttributeNS(ns string, name string) bool
+	Matches(string) bool
```

Comment:
Adding methods to an exported interface is technically a breaking API change (if anyone happened to implement this interface externally, their implementation would no longer satisfy it).

We have no choice but to allow this, because the Element interface is meant to map to the DOM Element interface, and the policy of this package allows for such API changes.